Test/Unit/Block/BlockTraitTest.php

```diff
@@ -0,0 +1,236 @@
+     *
+     * @param $checkoutKey
+     * @param $apiKey
+     * @param $signingSecretKey
+     * @param $expected
+     */
+    public function isKeyMissing($checkoutKey, $apiKey, $signingSecretKey, $expected)
+    {
+        $this->currentMock->expects(self::any())->method('getCheckoutKey')->willReturn($checkoutKey);
+        $this->configHelper->expects(self::any())->method('getApiKey')->willReturn($apiKey);
+        $this->configHelper->expects(self::any())->method('getSigningSecret')->willReturn($signingSecretKey);
+        $this->assertEquals($expected, $this->currentMock->isKeyMissing());
+    }
+
+    public function dataProvider_isKeyMissing()
+    {
+        return [
+            ['checkout_key', 'api_key', 'signing_secret_key', false],
+            ['checkout_key', 'api_key', '', true],
+            ['checkout_key', '', '', true],
+            ['', '', '', true],
+        ];
+    }
+
+
+    /**
+     * @test
+     * @dataProvider dataProvider_isEnabled
+     * @param $apiKey
+     * @param $expected
+     */
+    public function isEnabled($apiKey, $expected)
+    {
+        $this->currentMock->expects(self::any())->method('getStoreId')->willReturn(self::STORE_ID);
+        $this->configHelper->expects(self::any())->method('isActive')->with(self::STORE_ID)->willReturn($apiKey);
+        $this->assertEquals($expected, $this->currentMock->isEnabled());
+    }
+
+    public function dataProvider_isEnabled()
+    {
+        return [
+            [true, true],
+            [false, false]
+        ];
+    }
+
+    /**
+     * @test
+     */
+    public function getPageBlacklist()
+    {
+        $this->configHelper->expects(self::once())->method('getPageBlacklist')->willReturn(self::PAGE_BLACK_LIST);
+        $this->assertEquals(self::PAGE_BLACK_LIST, TestHelper::invokeMethod($this->currentMock, 'getPageBlacklist'));
+    }
+
+    /**
+     * @test
+     */
+    public function getPageWhitelist()
+    {
+        $this->configHelper->expects(self::once())->method('getPageWhitelist')->willReturn([self::PAGE_WHITE_LIST]);
+        $this->assertEquals([
+            Config::SHOPPING_CART_PAGE_ACTION,
+            Config::CHECKOUT_PAGE_ACTION,
+            Config::SUCCESS_PAGE_ACTION,
+            self::PAGE_WHITE_LIST
+        ], TestHelper::invokeMethod($this->currentMock, 'getPageWhitelist'));
+    }
+
+    /**
+     * @test
+     * @dataProvider dataProvider_isIPRestricted
+     * @param $expected
+     * @param $isIPRestricted
+     * @throws \ReflectionException
+     */
+    public function isIPRestricted($expected, $isIPRestricted)
+    {
+        $this->configHelper->expects(self::once())->method('isIPRestricted')->willReturn($isIPRestricted);
+        $this->assertEquals($expected, TestHelper::invokeMethod($this->currentMock, 'isIPRestricted'));
+    }
+
+    public function dataProvider_isIPRestricted()
+    {
+        return [
+            [true, true],
+            [false, false]
+        ];
+    }
+}
+
```

Comment:
@johnmccombs1 Similar to the other PR, tons of whitespace here.

Reply:
@matt-thomason I've updated the code